systemvm/patches/debian/config/opt/cloud/bin/cs/CsAddress.py

```diff
@@ -95,9 +95,25 @@ def get_control_if(self):
                 return ip
         return None
 
+    def check_if_link_exists(self,dev):
+        cmd="ip link show dev %s"%dev
+        result=CsHelper.execute(cmd)
+        if(len(result)!=0):
+           return True
+        else:
+           return False
+
+    def check_if_link_up(self,dev):
+        cmd="ip link show dev %s | tr '\n' ' ' | cut -d ' ' -f 9"%dev
```

Comment:
Hi, 
we have used the subprocess command to find the state of the device in two more places. I have created a improvement ticket(CLOUDSTACK-8837) to track this. I will work on this later, until then 
i guess the current changes will do.